src/Transports.AspNetCore/GraphQLHttpMiddlewareOptions.cs

```diff
@@ -0,0 +1,107 @@
+    /// to initialize the WebSocket pipeline within the ASP.NET Core framework.
+    /// </summary>
+    public bool HandleWebSockets { get; set; } = true;
+
+    ******************/
+
+    /// <summary>
+    /// Enables handling of batched GraphQL requests for POST requests when formatted as JSON.
+    /// </summary>
+    public bool EnableBatchedRequests { get; set; } = true;
+
+    /// <summary>
+    /// Enables parallel execution of batched GraphQL requests.
+    /// </summary>
+    public bool BatchedRequestsExecuteInParallel { get; set; } = true;
+
+    /// <summary>
+    /// When enabled, GraphQL requests with validation errors
+    /// have the HTTP status code set to 400 Bad Request.
+    /// GraphQL requests with execution errors are unaffected.
+    /// <br/><br/>
+    /// Does not apply to batched or WebSocket requests.
+    /// </summary>
+    public bool ValidationErrorsReturnBadRequest { get; set; } = true;
+
+    /// <summary>
+    /// Enables parsing the query string on POST requests.
+    /// If enabled, the query string properties override those in the body of the request.
+    /// </summary>
+    public bool ReadQueryStringOnPost { get; set; } = true;
+
+    /// <summary>
+    /// Enables reading variables from the query string.
+    /// Variables are interpreted as JSON and deserialized before being
+    /// provided to the <see cref="IDocumentExecuter"/>.
+    /// </summary>
+    public bool ReadVariablesFromQueryString { get; set; } = true;
+
+    /// <summary>
+    /// Enables reading extensions from the query string.
+    /// Extensions are interpreted as JSON and deserialized before being
+    /// provided to the <see cref="IDocumentExecuter"/>.
+    /// </summary>
+    public bool ReadExtensionsFromQueryString { get; set; } = true;
+
+    /// <summary>
+    /// If set, requires that <see cref="IIdentity.IsAuthenticated"/> return <see langword="true"/>
+    /// for the user within <see cref="HttpContext.User"/>
+    /// prior to executing the GraphQL request or accepting the WebSocket connection.
+    /// </summary>
+    public bool AuthorizationRequired { get; set; }
+
+    /// <summary>
+    /// Requires that <see cref="ClaimsPrincipal.IsInRole(string)"/> return <see langword="true"/>
+    /// for the user within <see cref="HttpContext.User"/>
+    /// for at least one role in the list prior to executing the GraphQL request or accepting
+    /// the WebSocket connection.  If no roles are specified, authorization is not checked.
+    /// </summary>
+    public List<string> AuthorizedRoles { get; set; } = new();
+
+    /// <summary>
+    /// If set, requires that <see cref="IAuthorizationService.AuthorizeAsync(ClaimsPrincipal, object, string)"/>
+    /// return a successful result for the user within <see cref="HttpContext.User"/>
+    /// for the specified policy before executing the GraphQL
+    /// request or accepting the WebSocket connection.
+    /// </summary>
+    public string? AuthorizedPolicy { get; set; }
```

Comment:
It should be remembered that the fields can be marked with own and different policies and principal should satisfy all of them during validation stage. So the question is - why to bother here with one more policy? Just one more (easy) way to require permissions **before** parsing GraphQL document?

Reply:
Read comment above.  It's for security to prevent unauthorized schema investigation of a secure endpoint.

Sorta like locking the outside doors to the shopping center even if all the doors inside the shopping center are locked tight.  It just provides a smaller attack surface to would-be attackers.

It's also incredibly easy to use -- but calling `this.Authorize()` within the schema constructor would be very easy to use too.

Reply:
Understand. OK.

> Sorta like locking the outside doors to the shopping center even if all the doors inside the shopping center are locked tight. It just provides a smaller attack surface to would-be attackers.

Very well said :).